apps/main/src/loan/components/PageLlamaMarkets/hooks/useLlamaMarketsColumnVisibility.tsx

```diff
@@ -13,50 +13,53 @@ export const createLlamaMarketsMobileColumns = (sortBy: LlamaMarketColumnId) =>
 /**
  * Create a map of column visibility for the Llama markets table that can be customized by the user.
  * This is not used on mobile devices (see `createLlamaMarketsMobileColumns` above).
- * @param isConnected Whether there is a wallet connected.
+ * @param hasPositions Whether the user is connected and has positions. Undefined during loading.
  */
-export const createLlamaMarketsColumnOptions = (isConnected: boolean) => [
-  {
-    label: t`Markets`,
-    options: [
-      {
-        label: t`Available Liquidity`,
-        columns: [LlamaMarketColumnId.LiquidityUsd],
-        active: true,
-        enabled: true,
-      },
-      {
-        label: t`Utilization`,
-        columns: [LlamaMarketColumnId.UtilizationPercent],
-        active: true,
-        enabled: true,
-      },
-    ],
-  },
-  {
-    label: t`Borrow`,
-    options: [
-      { columns: [LlamaMarketColumnId.BorrowRate], active: true, enabled: true },
-      { label: t`Chart`, columns: [LlamaMarketColumnId.BorrowChart], active: true, enabled: true },
-      {
-        label: t`Borrow Details`,
-        columns: [LlamaMarketColumnId.UserHealth, LlamaMarketColumnId.UserBorrowed],
-        active: true,
-        enabled: isConnected,
-      },
-    ],
-  },
-  {
-    label: t`Lend`,
-    options: [
-      { columns: [LlamaMarketColumnId.LendRate], active: true, enabled: true },
-      { label: t`Chart`, columns: [LlamaMarketColumnId.LendChart], active: false, enabled: true },
-      {
-        label: t`Lend Details`,
-        columns: [LlamaMarketColumnId.UserEarnings, LlamaMarketColumnId.UserDeposited],
-        active: true,
-        enabled: isConnected,
-      },
-    ],
-  },
-]
+export const createLlamaMarketsColumnOptions = (hasPositions: boolean | undefined) =>
+  hasPositions == null
```

Comment:
Is this ternary really needed? Previously, even when not connected wouldn't it still show the enabled columns (the first 2?)

Reply:
That's right, previously we had the `isConnected` which came from a different data source than the rest of the table (wagmi).
Now, the `hasPositions` boolean is available at the same time as the table data is loaded, since it's part of the table query